src/compaction.c

```diff
@@ -347,21 +351,87 @@ void LastAppendValue(void *contextPtr, double value) {
     context->isResetted = FALSE;
 }
 
-static AggregationClass aggMax = { .createContext = MaxMinCreateContext,
-                                   .appendValue = MaxMinAppendValue,
+void *RangeCreateContext() {
+    RangeContext *context = (RangeContext *)malloc(sizeof(RangeContext));
+    context->minValue = 0;
+    context->maxValue = 0;
+    context->isResetted = TRUE;
+    return context;
+}
+
+void RangeAppendValue(void *contextPtr, double value) {
+    RangeContext *context = (RangeContext *)contextPtr;
+    if (context->isResetted) {
+        context->isResetted = FALSE;
+        context->maxValue = value;
+        context->minValue = value;
+    } else {
+        if (value > context->maxValue) {
+            context->maxValue = value;
+        }
+        if (value < context->minValue) {
+            context->minValue = value;
+        }
+    }
+}
+
+int RangeMinFinalize(void *contextPtr, double *value) {
+    RangeContext *context = (RangeContext *)contextPtr;
+    if (context->isResetted == TRUE) {
+        return TSDB_ERROR;
+    }
+    *value = context->minValue;
+    return TSDB_OK;
+}
+
+int RangeFinalize(void *contextPtr, double *value) {
+    RangeContext *context = (RangeContext *)contextPtr;
+    if (context->isResetted == TRUE) {
+        return TSDB_ERROR;
+    }
+    *value = context->maxValue - context->minValue;
+    return TSDB_OK;
+}
+
+void RangeReset(void *contextPtr) {
+    RangeContext *context = (RangeContext *)contextPtr;
+    context->maxValue = 0;
+    context->minValue = 0;
+    context->isResetted = TRUE;
+}
+
+void RangeWriteContext(void *contextPtr, RedisModuleIO *io) {
+    RangeContext *context = (RangeContext *)contextPtr;
+    RedisModule_SaveDouble(io, context->maxValue);
+    RedisModule_SaveDouble(io, context->minValue);
+    RedisModule_SaveStringBuffer(io, &context->isResetted, 1);
+}
+
+void RangeReadContext(void *contextPtr, RedisModuleIO *io) {
```

Comment:
@filipecosta90 how are we handling RDB backwards compatibility for this change?

Reply:
nice catch. fixing it :)